File: core/emergency-manager/app/src/main/java/project/grp3/emergency/core/database/repositories/SensorRepository.java
package project.grp3.emergency.core.database.repositories;

import project.grp3.emergency.core.database.entities.SensorEntity;

import java.util.List;

public class SensorRepository extends Repository<SensorEntity>
{
    public SensorRepository()
    {
        super(SensorEntity.class);
    }

    public List<SensorEntity> getAll()
    {
        return super.getAll();
    }

    public SensorEntity getOneById(Long id){ return super.get(id); }

}

File: core/mqtt/app/src/database/repositories/SensorRepository.ts
import {EntityRepository,  Repository} from "typeorm";
import {SensorEntity} from "../entities/SensorEntity";
import {SensorTopicResponse} from "../../config/mqtt";


@EntityRepository(SensorEntity)
export class SensorRepository extends Repository<SensorEntity> {
    public getAll(): Promise<SensorEntity[]> {
        return this.find()
    }

    async add(data: SensorTopicResponse["data"]): Promise<SensorEntity>  {
        const entity = this.create({
            ...data,
            sensorId: data.id,
            id: undefined,
            date: new Date(),
            fires: []
        });
        return await this.getBySensorId(data.id) ?? await super.save(entity)
    }

    async getBySensorId(id: number): Promise<SensorEntity> {
        return (await this.find({where: {sensorId: id}}))[0];
    }


}
